# About Code
kode dibawah ini adalah kode yang saya buat untuk melakukan scrapping data ulasan aplikasi Triv, yang bersumber dari Google Playstore. kode ini akan menghasilkan data ulasan yang berisikan beberapa kolom yaitu :

1. reviewId - Id ulasan

2. userName - Username dari pengulas aplikasi

3. userImage - Foto rofil dari pengulas aplikasi

4. content - isi ulasan pengguna

5. score - Score bintang kepuasan pengguna

6. thumbsUpCount - Jumlah like ulasan

7. reviewCreatedVersion - Versi aplikasi ketika ulasan dibuat

8. at - Waktu ulasan dibuat

9. replyContent - balasaan dari pihak aplikasi

10. repliedAt - Waktu balasan

11. appVersion - versi aplikasi

In [1]:
# Mengimpor pustaka google_play_scraper untuk mengakses ulasan dan informasi aplikasi dari Google Play Store.
from google_play_scraper import app, reviews, Sort, reviews_all
import pandas as pd

In [4]:
scrapreview = reviews_all(
    'id.co.triv',       # ID aplikasi
    lang='id',               # Bahasa ulasan (default: 'en')
    country='id',            # Negara (default: 'us')
    sort=Sort.MOST_RELEVANT, # Urutan ulasan (default: Sort.MOST_RELEVANT)
    count=10000              # Jumlah maksimum ulasan yang ingin diambil
)  

In [6]:
len(scrapreview)

12665

In [7]:
df = pd.DataFrame(scrapreview)

In [8]:
df.head()

reviewId         userName  \
0  41dff01f-cf34-4128-9369-aad8cc1a0d2e  Pengguna Google   
1  b1fa53fe-fc80-4d7c-8fb4-7ada6ed46d0e  Pengguna Google   
2  05d3274d-d0ec-4777-86a3-d3259b91a224  Pengguna Google   
3  63d8f291-045f-4e47-93ab-cc069aad1425  Pengguna Google   
4  bb4df9c8-3276-4259-aa86-526399a86fdb  Pengguna Google   

                                           userImage  \
0  https://play-lh.googleusercontent.com/EGemoI2N...   
1  https://play-lh.googleusercontent.com/EGemoI2N...   
2  https://play-lh.googleusercontent.com/EGemoI2N...   
3  https://play-lh.googleusercontent.com/EGemoI2N...   
4  https://play-lh.googleusercontent.com/EGemoI2N...   

                                             content  score  thumbsUpCount  \
0  Triv adalah aplikasi yang luar biasa untuk ber...      5             18   
1  Saya sudah menggunakan Triv selama beberapa bu...      5             10   
2  Saya sudah menggunakan Triv selama beberapa bu...      5              2   
3  Aplikasi Triv sangat membantu untuk jual beli ...      5              8   
4  Udah pakai Triv selama beberapa bulan dan sang...      5              0   

  reviewCreatedVersion                  at replyContent repliedAt appVersion  
0                8.0.4 2025-06-21 01:28:23         None       NaT      8.0.4  
1                8.0.4 2025-06-19 14:58:26         None       NaT      8.0.4  
2                8.0.4 2025-06-20 12:19:38         None       NaT      8.0.4  
3                8.0.4 2025-06-21 06:22:09         None       NaT      8.0.4  
4                8.0.4 2025-06-18 09:50:33         None       NaT      8.0.4

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12665 entries, 0 to 12664
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   reviewId              12665 non-null  object        
 1   userName              12665 non-null  object        
 2   userImage             12665 non-null  object        
 3   content               12665 non-null  object        
 4   score                 12665 non-null  int64         
 5   thumbsUpCount         12665 non-null  int64         
 6   reviewCreatedVersion  9841 non-null   object        
 7   at                    12665 non-null  datetime64[ns]
 8   replyContent          401 non-null    object        
 9   repliedAt             401 non-null    datetime64[ns]
 10  appVersion            9841 non-null   object        
dtypes: datetime64[ns](2), int64(2), object(7)
memory usage: 1.1+ MB


Dataset ini berisi 12665  entri ulasan aplikasi dengan 11 kolom. Semua ulasan memiliki reviewId, userName, userImage, content, score, dan thumbsUpCount yang lengkap. Tanggal ulasan (at) juga tersedia untuk semua entri. Namun, beberapa kolom memiliki data yang tidak lengkap, seperti reviewCreatedVersion dan appVersion, yang hanya terisi hanya 9841 baris. Selain itu, hanya 401 ulasan yang mendapat balasan dari pengembang, ditunjukkan oleh replyContent dan repliedAt. Hal ini menunjukkan bahwa sebagian besar ulasan tidak memiliki tanggapan resmi dari pihak aplikasi.

In [10]:
df.to_csv('data/triv.csv', index=False)